Source/WebKit/UIProcess/API/Swift/WebPage.swift

```diff
@@ -39,16 +106,55 @@ import Foundation
 
     public init() {
         let (stream, continuation) = AsyncStream.makeStream(of: NavigationEvent.self)
-        self.navigations = Navigations(source: stream)
-        self.backingNavigationDelegate = WKNavigationDelegateAdapter(navigationProgressContinuation: continuation)
+        navigations = Navigations(source: stream)
```

Comment:
I see this isn't new here, but something to consider is whether we want to have a single value here or if the getter for `navigations` should return a fresh sequence every time. I think this ended up being a `struct`, so you are getting an implicitly `Copyable` value for now, but `AsyncSequence` itself doesn't specify whether a sequence may be consumed multiple times or not. We should figure out our behaviors here and make sure we've got an implementation to match.

Reply:
I added a FIXME to consider and address this